src/types/reference.ts

```diff
@@ -16,7 +16,10 @@ export interface IReferenceDescription {
 
 export function referenceTo<T>(path: string): T;
 export function referenceTo<T>(getter: IReferenceGetter<T>, setter?: IReferenceSetter<T>): T;
+export function referenceTo<T>(factory: IFactory<any, T>): T;
 export function referenceTo(arg1, arg2?) {
```

Comment:
TODO: Maybe export as "reference" to be consistent with other types and more readable?